qml/Keyboard.qml

```diff
@@ -292,19 +308,46 @@ Item {
             }
             onThemeChanged: Theme.load(target.theme)
         }
+        
+        FloatingActions {
+            id: floatingActions
+            objectName: "floatingActions"
+            
+            z: 1
+            visible: fullScreenItem.cursorSwipe && !cursorSwipeArea.pressed
+        }
 
         MouseArea {
             id: cursorSwipeArea
-            anchors.fill: parent
+            
+            property point firstPress
+            property point lastRelease
+            property bool selectionMode: false
+            
+            anchors {
+                fill: parent
+                topMargin: toolbar.height
+            }
+            
             enabled: cursorSwipe
 
             Rectangle {
                 anchors.fill: parent
                 visible: parent.enabled
-                color: fullScreenItem.theme.charKeyPressedColor
-                opacity: 0.5
+                color: cursorSwipeArea.selectionMode ? fullScreenItem.theme.selectionColor : fullScreenItem.theme.charKeyPressedColor
```

Comment:
We shouldn't distinguish the mode only by the color of the keyboard background. This helps people who aren't able to distinguish between blue and their primary keyboard color.

Should we add some text to the keyboard, such as "Drag here to move cursor" or "Drag here to move selection"?

Reply:
I will try to add some texts for both modes.

Reply:
I've added texts in the latest commit.